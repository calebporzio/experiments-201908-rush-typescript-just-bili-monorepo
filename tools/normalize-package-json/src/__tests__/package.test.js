import path from 'path';

import { Package } from '..';

const fixture = name => ({
  rootDir: path.resolve(__dirname, 'fixtures', name),
});

describe('normalize-package-json', () => {
  test('autoFix should sort', () => {
    const opts = fixture('package-unsorted');
    const subject = new Package({ ...opts, sortDependencies: true });
    subject.autoFix();
    expect(subject.toString()).toMatchSnapshot();
  });
});
