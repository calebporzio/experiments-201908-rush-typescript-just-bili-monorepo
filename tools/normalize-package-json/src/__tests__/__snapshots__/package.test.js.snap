// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalize-package-json autoFix should sort 1`] = `
"{
  \\"name\\": \\"package-unsorted\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"license\\": \\"MIT\\",
  \\"bin\\": {
    \\"test\\": \\"test\\"
  },
  \\"dependencies\\": {}
}
"
`;
